### ISSUE TITLE
[BUG] Failed to update the app after downloading it

### ISSUE THREAD

**Issue by MesterPerfect** [marticliment/WingetUI#1521]
### Please confirm these before moving forward

- [X] I have searched for my issue and not found a work-in-progress/duplicate/resolved issue.
- [X] I have tested that this issue has not been fixed in the latest [(beta or stable) release](https://github.com/marticliment/WingetUI/releases/).
- [X] I have checked the [FAQ](https://github.com/marticliment/WingetUI#frequently-asked-questions) section for solutions
- [X] This issue is about a bug (if it is not, please use the [correct template](https://github.com/marticliment/WingetUI/issues/new/choose))

### Describe your issue

I am using the latest version of WingetUI 2.1.1. There is a problem with many programs that occurs when I try to update them. For example, if I click on Update and start downloading and installing TeamViewer, it fails completely. This happens with many other applications as well. This is the log of the operation.

### Steps to reproduce the issue

1. Open WingetUI.
2. Click the Updates tab.
3. Select an application that fails to update.
4. Right-click the application and select Update as administrator.

### WingetUI Log

```text

🟢 Total packages: 4
🔵 Starting Winget search for installed packages
🔵 Starting Chocolatey search for installed packages
🟢 Searching for string ""
🟢 Chocolatey search for installed packages finished with 0 result(s)
🟠 Chocolatey got too few installed packages, retrying
🔵 Starting Chocolatey search for installed packages
🟢 Chocolatey search for installed packages finished with 0 result(s)
🟢 Searching for string ""
🟢 Waiting for install permission... title=TeamViewer, id=TeamViewer.TeamViewer, installId=1698183973.9369469
🔵 Given package: <Package: TeamViewer;TeamViewer.TeamViewer;15.46.7;Winget: winget;<PackageManagers.winget.WingetPackageManager object at 0x00000141E53C7790>;<Interface.CustomWidgets.SpecificWidgets.UpgradablePackageItem object at 0x00000141E6E00900>>
🔵 Installation options: <InstallationOptions: SkipHashCheck=False;InteractiveInstallation=False;RunAsAdministrator=True;Version=;Architecture=;InstallationScope=;CustomParameters=[];RemoveDataOnUninstall=False>
🔵 Found icon:  https://i.imgur.com/2UWkAjM.png
Win32Mica Error: win32 version 19045 is not supported
🔵 Current program set to 1698183973.9369469
🟢 Have permission to install, starting installation threads...
🔵 Starting <Package: TeamViewer;TeamViewer.TeamViewer;15.46.7;Winget: winget;<PackageManagers.winget.WingetPackageManager object at 0x00000141E53C7790>;<Interface.CustomWidgets.SpecificWidgets.UpgradablePackageItem object at 0x00000141E6E00900>> update with Command ['C:\\Program Files\\WingetUI\\components\\gsudo.exe', 'C:\\Program Files\\WingetUI\\PackageManagers\\winget-cli_x64\\winget.exe', 'upgrade', '--id', 'TeamViewer.TeamViewer', '--exact', '--include-unknown', '--accept-source-agreements', '--disable-interactivity', '--accept-package-agreements']
🟢 Winget search for installed packages finished with 138 result(s)
🟢 Searching for string ""
🟢 Searching for string ""
🟢 Total packages: 138
🔵 Starting Winget search for installed packages
🔵 Starting Chocolatey search for installed packages
🟢 Searching for string ""
🟢 Chocolatey search for installed packages finished with 0 result(s)
🟠 Chocolatey got too few installed packages, retrying
🔵 Starting Chocolatey search for installed packages
🟢 Chocolatey search for installed packages finished with 0 result(s)
🟢 Searching for string ""
🟢 Winget search for installed packages finished with 138 result(s)
🟢 Searching for string ""
🟢 Searching for string ""
🟢 Total packages: 138
.\PackageManagers\winget.py:638: RuntimeWarning: libshiboken: Overflow: Value 2316632070 exceeds limits of type  [signed] "int" (4bytes).
OverflowError
['-', '\\', '|', '███▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  1024 KB / 8.23 MB', '███████▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  2.00 MB / 8.23 MB', '██████████▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  3.00 MB / 8.23 MB', '██████████████▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  4.00 MB / 8.23 MB', '██████████████████▒▒▒▒▒▒▒▒▒▒▒▒  5.00 MB / 8.23 MB', '█████████████████████▒▒▒▒▒▒▒▒▒  6.00 MB / 8.23 MB', '█████████████████████████▒▒▒▒▒  7.00 MB / 8.23 MB', '█████████████████████████████▒  8.00 MB / 8.23 MB', '██████████████████████████████  8.23 MB / 8.23 MB', '-', '\\', 'Found TeamViewer [TeamViewer.TeamViewer] Version 15.47.3', 'This application is licensed to you by its owner.', 'Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.', 'Downloading https://download.teamviewer.com/download/TeamViewer_Setup_x64.exe', '-', '\\', '|', '/', '-', '\\', '|', '/', '▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  0.00 B / 59.4 MB', '▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  1024 KB / 59.4 MB', '██▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  5.00 MB / 59.4 MB', '████▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  8.00 MB / 59.4 MB', '██████▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  12.0 MB / 59.4 MB', '██████▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  13.0 MB / 59.4 MB', '████████▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  16.0 MB / 59.4 MB', '██████████▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  21.0 MB / 59.4 MB', '████████████▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  24.0 MB / 59.4 MB', '█████████████▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  27.0 MB / 59.4 MB', '███████████████▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  30.0 MB / 59.4 MB', '████████████████▒▒▒▒▒▒▒▒▒▒▒▒▒▒  33.0 MB / 59.4 MB', '██████████████████▒▒▒▒▒▒▒▒▒▒▒▒  36.0 MB / 59.4 MB', '███████████████████▒▒▒▒▒▒▒▒▒▒▒  39.0 MB / 59.4 MB', '█████████████████████▒▒▒▒▒▒▒▒▒  42.0 MB / 59.4 MB', '██████████████████████▒▒▒▒▒▒▒▒  44.0 MB / 59.4 MB', '████████████████████████▒▒▒▒▒▒  48.0 MB / 59.4 MB', '█████████████████████████▒▒▒▒▒  50.0 MB / 59.4 MB', '██████████████████████████▒▒▒▒  53.0 MB / 59.4 MB', '███████████████████████████▒▒▒  55.0 MB / 59.4 MB', '████████████████████████████▒▒  56.4 MB / 59.4 MB', '██████████████████████████████  59.4 MB / 59.4 MB', '██████████████████████████████  59.4 MB / 59.4 MB', 'Successfully verified installer hash', 'Starting package install...', '-', '\\', '|', '/', '-', '\\', '|', '/', '-', '\\', '|', '/', '-', '\\', 'Installer failed with exit code: 2', '']
Win32Mica Error: win32 version 19045 is not supported
Win32Mica Error: win32 version 19045 is not supported

```

### Package Managers Logs

```text
-
\
|
███▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  1024 KB / 8.23 MB
███████▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  2.00 MB / 8.23 MB
██████████▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  3.00 MB / 8.23 MB
██████████████▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  4.00 MB / 8.23 MB
██████████████████▒▒▒▒▒▒▒▒▒▒▒▒  5.00 MB / 8.23 MB
█████████████████████▒▒▒▒▒▒▒▒▒  6.00 MB / 8.23 MB
█████████████████████████▒▒▒▒▒  7.00 MB / 8.23 MB
█████████████████████████████▒  8.00 MB / 8.23 MB
██████████████████████████████  8.23 MB / 8.23 MB
-
\
Found TeamViewer [TeamViewer.TeamViewer] Version 15.47.3
This application is licensed to you by its owner.
Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
Downloading https://download.teamviewer.com/download/TeamViewer_Setup_x64.exe
-
\
|
/
-
\
|
/
▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  0.00 B / 59.4 MB
▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  1024 KB / 59.4 MB
██▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  5.00 MB / 59.4 MB
████▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  8.00 MB / 59.4 MB
██████▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  12.0 MB / 59.4 MB
██████▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  13.0 MB / 59.4 MB
████████▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  16.0 MB / 59.4 MB
██████████▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  21.0 MB / 59.4 MB
████████████▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  24.0 MB / 59.4 MB
█████████████▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  27.0 MB / 59.4 MB
███████████████▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  30.0 MB / 59.4 MB
████████████████▒▒▒▒▒▒▒▒▒▒▒▒▒▒  33.0 MB / 59.4 MB
██████████████████▒▒▒▒▒▒▒▒▒▒▒▒  36.0 MB / 59.4 MB
███████████████████▒▒▒▒▒▒▒▒▒▒▒  39.0 MB / 59.4 MB
█████████████████████▒▒▒▒▒▒▒▒▒  42.0 MB / 59.4 MB
██████████████████████▒▒▒▒▒▒▒▒  44.0 MB / 59.4 MB
████████████████████████▒▒▒▒▒▒  48.0 MB / 59.4 MB
█████████████████████████▒▒▒▒▒  50.0 MB / 59.4 MB
██████████████████████████▒▒▒▒  53.0 MB / 59.4 MB
███████████████████████████▒▒▒  55.0 MB / 59.4 MB
████████████████████████████▒▒  56.4 MB / 59.4 MB
██████████████████████████████  59.4 MB / 59.4 MB
██████████████████████████████  59.4 MB / 59.4 MB
Successfully verified installer hash
Starting package install...
-
\
|
/
-
\
|
/
-
\
|
/
-
\
Installer failed with exit code: 2
```

### Screenshots and videos

![image](https://github.com/marticliment/WingetUI/assets/44323153/1a9faa7f-eb74-4f34-8f6f-ffab234ee097)

![image](https://github.com/marticliment/WingetUI/assets/44323153/bb86ece4-516b-4ebf-b22c-b32add0ce9c9)

**Comment by marticliment** [comment]
does running "winget upgrade teamviewer" work?

**Comment by MesterPerfect** [comment]
> does running "winget upgrade teamviewer" work?

I needed to update it manually, but three packages still refused to upgrade.

![image](https://github.com/marticliment/WingetUI/assets/44323153/64375f0c-e633-4ca9-b91a-2e068eb3d456)


![image](https://github.com/marticliment/WingetUI/assets/44323153/64b58457-bdbe-42d8-8602-921fb5b15a88)

**Comment by MesterPerfect** [comment]
Everything is up to date 

![image](https://github.com/marticliment/WingetUI/assets/44323153/16cb5ab3-d67d-42ae-b457-48543ee37e3e)


i found that the TeamViewer upgrade failed because I was using the x86 version of TeamViewer, while winget was trying to install the x64 version. If we try to install via the GUI, the TeamViewer installation wizard displays a message stating that the x86 version must be removed in order for the x64 version to be installed. I don't think this happens with an unattended installation.

 **Suggested solution:** If winget-ui can access the path of files downloaded by winget-cli, then if the installation fails, a message can be shown to the user informing them that the installation has failed and asking them if they want to try installing the program manually.

**Comment by marticliment** [comment]
> i found that the TeamViewer upgrade failed because I was using the x86 version of TeamViewer, while winget was trying to install the x64 version. If we try to install via the GUI, the TeamViewer installation wizard displays a message stating that the x86 version must be removed in order for the x64 version to be installed. I don't think this happens with an unattended installation.

This is a known issue with TeamViewer.



> **Suggested solution:** If winget-ui can access the path of files downloaded by winget-cli, then if the installation fails, a message can be shown to the user informing them that the installation has failed and asking them if they want to try installing the program manually.

WingetUI does not have access to the installer paths

**Comment by MesterPerfect** [comment]
> This is a known issue with TeamViewer.

Yes, it is an annoying issue

> WingetUI does not have access to the installer paths

Therefore, you cannot handle errors of this type.

**Comment by marticliment** [comment]
> Therefore, you cannot handle errors of this type.

Exactly

**Comment by alphanightshift** [comment]
> Everything is up to date
> 
> ![image](https://user-images.githubusercontent.com/44323153/278060739-16cb5ab3-d67d-42ae-b457-48543ee37e3e.png)
> 
> i found that the TeamViewer upgrade failed because I was using the x86 version of TeamViewer, while winget was trying to install the x64 version. If we try to install via the GUI, the TeamViewer installation wizard displays a message stating that the x86 version must be removed in order for the x64 version to be installed. I don't think this happens with an unattended installation.
> 
> **Suggested solution:** If winget-ui can access the path of files downloaded by winget-cli, then if the installation fails, a message can be shown to the user informing them that the installation has failed and asking them if they want to try installing the program manually.

How did you update those desktop runtime with different architechtures (x86 and x64) on wingetui? Every time I try to update them it always fails since they cannot find the input that matches criteria or something, therefore I have to access the winget CLI to update them.

**Comment by marticliment** [comment]
This is an issue that was caused by a bug in winget 1.6, that has been fixed on winget 1.7. Therefore, this bug may happen until wingetui adopts winget 1.7
